### PR TITLE
Fix #101 - support for mailto links

### LINKED ISSUE #101: Mailto links not supported in firefox
## Expected Behavior
Firefox shows a small pop-up in top right hand corner saying "Add protonmail as a mailto application"

## Actual Behavior
No such popup

## Steps to Reproduce the Problem

  1. Open protonmail.com
  2. Login
  3. Try and search for firefox popup to add as mailto.

## Specifications

  - Version:
  - Platform: FIrefox
  - URL: Protonmail.com

P.S Mailto links in firefox are supported by tutanota:

https://mail.tutanota.com/login?requestedPath=%2Fmailto%23url%3Dmailto%253Atnp%2540schabi.org

You can see mailto link in the browser url. For more privacy maybe you could obfuscate the url too

### PR DESCRIPTION
**Pull request by ArcanisCz**
Hi!
Added support for mailto links from #101. Three weak points though:
- only basic mailto support. With extended support, i ran into url escaping issues. At that point, it would be best to add separate route for that, with its own query params to not mix with conversation query params. And that would be quite big addition in app (With extended support, i mean `/inbox?mailtoUrl=mailto:name1@rapidtables.com%3Fcc%3Dname2@rapidtables.com&bcc=name3@rapidtables.com&subject=The%20subject%20of%20the%20email&body=The%20body%20of%20the%20email`)
- mailtoUrl is scattered in multiple places, since it has to be retained through login. Also, it would be best if the name is some constant.
- No tests added and no tests ran, because your stack is heavily dependant on unix based systems, and i have windows. As for yet, i have no quick solution for running tests on my pc.

Hope it helps!
Arcanis

### CODE REVIEW

File: src/app/app.js [hunk 1]
```diff
@@ -82,6 +82,8 @@ angular
 
         // SVG Polyfill for IE11 @todo lazy load
         window.svg4everybody();
+
+        navigator.registerProtocolHandler("mailto", "https://mail.protonmail.com/inbox?mailtoUrl=%s", "Protonmail");
```
Comment (ArcanisCz) [inline src/app/app.js]: maybe add some if for not triggering in dev enviroment?

File: src/app/app.js [hunk 2]
```diff
@@ -82,6 +82,8 @@ angular
 
         // SVG Polyfill for IE11 @todo lazy load
         window.svg4everybody();
+
+        navigator.registerProtocolHandler("mailto", "https://mail.protonmail.com/?mailtoUrl=%s", "Protonmail");
```
Comment (ArcanisCz) [inline src/app/app.js:86]: maybe add some if for not triggering in dev enviroment? Also, hardcoded URL is bad - but i dont know much about your deployment process and enviroments.

**Comment by ArcanisCz** [comment]
Any news about this PR after few months? Are you interested or should i close it?

**Comment by dhoko** [comment]
> Any news about this PR after few months? Are you interested or should i close it?

Hey 👋  sorry for the delay :/ 
We've talked about it and that's a good idea but it would be better if we were able to only load the composer instead of the whole app. ex: Something similar to an encrypted email sent to a non protonmail account. 

As it's done today it would be difficult to do, we need to load the whole app :/ We added this feature inside the pipe, it should come later.

**Comment by ArcanisCz** [comment]
Ok, no problem, only tried to fix issue #101 

Maybe you could consider to share your issue to work on, if you deem project as opensource? This way (and given protonmail's response times), its hard to contribute....

**Comment by stephanlachnit** [comment]
> We added this feature inside the pipe, it should come later.

Are you still working on this?

**Comment by dhoko** [comment]
Yes, but it's not high priority.
We need to make changes inside the webapp before, because having to load the full webclient is a bit too much.

**Comment by stephanlachnit** [comment]
I don't know about the technical details, but would it be possible to simply save the mailto link locally one the users machine, and parse it to the webapp once the user is logged in?

I think ProtonMail is great for everyone who cares about FOSS, but a lot of FOSS development is done with mail and the mailto protocol helps to easily reply to messages. For example take the Debian BTS: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=923851

You can reply with one click to messages thanks to the mailto links. On firefox it is pretty simple to write extensions which add mailto support for websites. I already created one for ProtonMail: https://github.com/stephanlachnit/mailto_protonmail (which obviously doesn't work yet)

**Pull request closed without merging**